app/models/user_block.rb

```diff
@@ -27,6 +27,12 @@ def active?
   end
 
   ##
+  # returns true if the block is a "zero hour" block
+  def zero_hour?
+    needs_view && (ends_at.to_i == updated_at.to_i)
```

Comment:
That's because `ends_at` is set by the controller by `created_at` and `updated_at` are either set by rails when the object is saved or by the database - not sure which. In any case truncation to seconds is not going to work if we happen to roll over a second boundary between the two...

Is `needs_view` not what you really want to check?